packages/graphql/lib/src/links/websocket_link/websocket_client.dart

```diff
@@ -239,17 +244,34 @@ class SocketClient {
       // Even though config.connect is sync, we call async in order to make the
       // SocketConnectionState.connected attribution not overload SocketConnectionState.connecting
       var connection =
-          await config.connect(uri: Uri.parse(url), protocols: protocols);
+          await config.connect(uri: Uri.parse(url), protocols: [protocol]);
       socketChannel = connection.forGraphQL();
       _connectionStateController.add(SocketConnectionState.connected);
       _write(initOperation);
 
       if (config.inactivityTimeout != null) {
-        _disconnectOnKeepAliveTimeout(_messages);
+        if (protocol == SocketSubProtocol.graphqlWs) {
+          _disconnectOnKeepAliveTimeout(_messages);
+        }
+        if (protocol == SocketSubProtocol.graphqlTransportWs) {
```

Comment:
All these conditionals are making the code harder to maintain and reason about. If we could refactor the protocol strategy out somehow, e.g. in a subclass, that would be stellar!

Reply:
Yeah, this is was I was thinking! I'm thinking only about how to divide the strategy pattern to include also a state pattern that manages the connection, so in this case, we can avoid all the if-else inside the class.

If the author of the PR have not time, I will take this task

Reply:
@vincenzopalazzo Right now I'm pretty stomped at work and don't have much time. Feel free to pick up where I left off!

Reply:
@maximilianmaihoefner ok! thanks for your initial work and sorry if I start to review this so late!

Reply:
>All these conditionals are making the code harder to maintain and reason about. If we could refactor the protocol strategy out somehow, e.g. in a subclass, that would be stellar!

@budde377 do you think that this change request can be a follow PR, in this way we can put in beta the new protocol support? and in the meanwhile working in the refactoring